backend/routes/gallery/canvas.js

```diff
@@ -32,23 +32,19 @@ export const saveCanvasData = async (canvasData) => {
         console.error("Error inserting canvas data:", e.stack || e);
         throw new Error("Failed to insert canvas data");
     } finally {
-        if (db) db.client.close(); // Close the database connection
+        if (db) db.client.close();
     }
 };
 
+// Get all canvases collection
 export const getAllCanvases = async () => {
     let db;
     try {
         db = await connectToDB();
-        const collection = db.collection(collectionName);
-
-        // Retrieve all canvases from the collection
-        const canvases = await collection.find().toArray();
-        return canvases; // Return the array of canvases
+        const collection = db.collection(collectionName); 
+        return collection; 
     } catch (e) {
         console.error("Error retrieving canvases:", e.stack || e);
         throw new Error("Failed to retrieve canvases");
-    } finally {
-        if (db) db.client.close();
```

Comment:
This is an error it leaks database connections. Put it back and add a comment

Reply:
It's still missing

Reply:
I moved that logic in paginatedResults.js